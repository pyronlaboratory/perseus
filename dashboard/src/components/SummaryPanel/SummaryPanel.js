import React, { useState } from 'react';

// material imports
import { makeStyles } from '@material-ui/core/styles';

import Accordion from '@material-ui/core/Accordion';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Divider from '@material-ui/core/Divider';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: 600,
        padding: theme.spacing(1, 0)
    },
    details:{
        fontSize: theme.typography.pxToRem(15),
        fontWeight: 400
    },
    divider: {
        background: '#dadada',
        margin: theme.spacing(1.5, 0),
        height: 0.5

    }
}));

const SummaryPanel = (props) => {

    const classes = useStyles();

    const [expanded, setExpanded] = useState(false);

    const handleChange = (panel) => (event, isExpanded) => {
        setExpanded(isExpanded ? panel : false);
    };

    return (
    
            <Accordion expanded={expanded === props.data.id} onChange={handleChange(props.data.id)}>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography className={classes.heading}>{props.data.user_lastname +", "+ props.data.user_firstname}</Typography>
            </AccordionSummary>
            {/* TO DO: Refactor details section */}
            <AccordionDetails style={{flexDirection: 'column'}}>
                <Typography className={classes.details} color="textSecondary" component="span">
                    {props.titleKey? props.data[props.titleKey]: null} 
                </Typography>
                <Divider className={classes.divider} />
                <Typography className={classes.secondaryDetails} color="textSecondary" component="span">
                    <strong style={{fontSize: "14px", color: '#66cbf9'}}>{props.subHeader}:</strong> <br /> 
                    {
                        props.subHeader === "Completion time" ?
                        Math.round(((props.data.course_completion_time)/36000) * 10) / 10 + ` hrs` :
                        
                        props.subHeader === "Submitted at" ? 
                        new Date(props.data.course_finished_at).toUTCString() :
                        props.data.total_courses_completed
                    }
                </Typography>
                
            </AccordionDetails>
            </Accordion>
    
    );
}

export default SummaryPanel