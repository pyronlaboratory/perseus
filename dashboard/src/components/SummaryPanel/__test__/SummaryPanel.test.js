import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import SummaryPanel from './../SummaryPanel';

afterEach(cleanup);

describe('summary panel component', () => {
    it("renders without crashing", () => {
        const mockData = {
            fastestSubmissions: [{"Name":"Randy Marsh"}],
            titleKey: "Tegridy Farms",
            subHeader: "Sales",
            key: 0
        }
        const div = document.createElement("div");
        ReactDOM.render(<SummaryPanel key={mockData.key} data={mockData.fastestSubmissions} titleKey={mockData.titleKey} subHeader={mockData.subHeader}/>, div);
    })

    it("matches snapshot", () => {
        const mockData = {
            fastestSubmissions: [{"Name":"Randy Marsh"}],
            titleKey: "Tegridy Farms",
            subHeader: "Sales"
        }

        const tree = renderer.create(<SummaryPanel key={mockData.key} data={mockData.fastestSubmissions} titleKey={mockData.titleKey} subHeader={mockData.subHeader} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})