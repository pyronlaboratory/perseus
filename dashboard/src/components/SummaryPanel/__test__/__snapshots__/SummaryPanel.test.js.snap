// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary panel component matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root makeStyles-heading-1 MuiTypography-body1"
      >
        undefined, undefined
      </p>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiAccordionDetails-root"
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <span
              className="MuiTypography-root makeStyles-details-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
            />
            <hr
              className="MuiDivider-root makeStyles-divider-3"
            />
            <span
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              <strong
                style={
                  Object {
                    "color": "#66cbf9",
                    "fontSize": "14px",
                  }
                }
              >
                Sales
                :
              </strong>
               
              <br />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
